ambry-api/src/main/java/com/github/ambry/named/NamedBlobDb.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+
+package com.github.ambry.named;
+
+import com.github.ambry.account.Container;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+
+public interface NamedBlobDb {
+  CompletableFuture<NamedBlobRecord> get(String accountName, String containerName, String blobName);
+
+  CompletableFuture<Page<NamedBlobRecord>> list(String accountName, String containerName, String blobNamePrefix, String continuationToken);
+
+  CompletableFuture<Void> put(NamedBlobRecord record);
+
+  CompletableFuture<Void> delete(String accountName, String containerName, String blobName);
```

Comment:
Will the caller ever need to know if the deletion happened?  Or will this throw an exception on blob not found?

Reply:
After, my most recent update, If this returns a success, the deletion will have taken place, or we found that the blob does not exist or was already deleted. I used to have it throw an exception on not found, but i don't think that is really needed since the deletion of the ambry blob would already tell us that. If there is a reason to know that info in the future, I can look at ways of changing this query. i.e. saving info to a variable and then selecting the variable afterwards.